crates/core/tedge_api/src/health.rs

```diff
@@ -104,42 +96,76 @@ impl ServiceHealthTopic {
     }
 }
 
-#[derive(Debug)]
-pub struct HealthTopicError;
-
 #[derive(Deserialize, Serialize, Debug, Default)]
 pub struct HealthStatus {
-    #[serde(default = "default_status")]
-    pub status: String,
+    #[serde(default)]
+    pub status: Status,
 }
```

Comment:
Is the `HealthStatus` structure still useful? `Status` seems enough.

At first I thought that the point was to serialize the `Status` along uptime and process id into MQTT messages. However, I noticed that this serialization is hard-coded by the `down_message` and `up_message` methods.

Reply:
That struct is still used to parse the json representation of it: `{ "status": "up" }`. And it also wraps the switching between the JSON version and the numeric version of payloads.